packages/near-membrane-base/src/membrane.ts

```diff
@@ -448,64 +455,51 @@ export function createMembraneMarshall() {
-                    // TODO: this might be problematic, because functions
-                    // and arrow functions should also be subject to this,
-                    // but it seems that we can still create a proxy of
-                    // a revoke, and wait until the user-land code actually
-                    // access something out of it to throw the proper error.
-                    // target is a revoked proxy, so the type doesn't matter
-                    // much from this point on
-                    targetTraits |= TargetTraits.Revoked;
-                }
-                targetTraits |= +targetIsArray && TargetTraits.IsArray;
-                targetTraits |= +!targetIsArray && TargetTraits.IsObject;
+                targetTraits |= TargetTraits.IsObject;
             }
             return targetTraits;
         }
 
         function getTargetIntegrityTraits(target: object): TargetIntegrityTraits {
             let targetIntegrityTraits = TargetIntegrityTraits.None;
             try {
-                // a revoked proxy will break the membrane when reading the meta
+                isArrayOrNotOrThrowForRevoked(target);
+            } catch {
+                targetIntegrityTraits |= TargetIntegrityTraits.Revoked;
+                return targetIntegrityTraits;
+            }
+            // A target may be a proxy that throws in its "isExtensible" trap.
+            try {
                 if (ObjectIsFrozen(target)) {
                     targetIntegrityTraits |=
-                        TargetIntegrityTraits.IsSealed &
                         TargetIntegrityTraits.IsFrozen &
+                        TargetIntegrityTraits.IsSealed &
                         TargetIntegrityTraits.IsNotExtensible;
                 } else if (ObjectIsSealed(target)) {
-                    targetIntegrityTraits |= TargetIntegrityTraits.IsSealed;
+                    targetIntegrityTraits |=
+                        TargetIntegrityTraits.IsSealed & TargetIntegrityTraits.IsNotExtensible;
                 } else if (!ReflectIsExtensible(target)) {
                     targetIntegrityTraits |= TargetIntegrityTraits.IsNotExtensible;
                 }
-                // if the target was revoked or become revoked during the extraction
-                // of the metadata, we mark it as broken in the catch.
-                isArrayOrNotOrThrowForRevoked(target);
-            } catch {
-                // intentionally swallowing the error because this method is just
-                // extracting the metadata in a way that it should always succeed
-                // except for the cases in which the target is a proxy that is
-                // either revoked or has some logic that is incompatible with the
-                // membrane, in which case we will just create the proxy for the
-                // membrane but revoke it right after to prevent any leakage.
-                targetIntegrityTraits |= TargetIntegrityTraits.Revoked;
-            }
```

Comment:
Yes this fixed the TODO we had distinguishing revoked from others.

Reply:
the only reason the `isArrayOrNotOrThrowForRevoked(target);` was the last line was to prevent them from revoking after the traits are produced. But I understand that this is not super super helpful anyways. So, I'm fine with this change.